src/EditorFeatures/CSharpTest/EditAndContinue/StatementMatchingTests.cs

```diff
@@ -863,6 +863,323 @@ public void LocalFunctionDefinitions()
+";
+            var src2 = @"
+object localF(object u) => u;
+a += localF;
+";
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "a += async u => u;", "a += localF;" },
+                { "async u => u", "object localF(object u) => u;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions4()
+        {
+            var src1 = @"
+bool condition(int l) { return l > 10; };
+foreach (var a in z)
+{
+    var e = from q in a.Where(condition) select q + 1;
+}
+";
+            var src2 = @"
+foreach (var a in z)
+{
+    var e = from q in a.Where(l => l < 0) select q + 1;
+}
+";
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "bool condition(int l) { return l > 10; }", "l => l < 0" },
+                { "foreach (var a in z) {     var e = from q in a.Where(condition) select q + 1; }", "foreach (var a in z) {     var e = from q in a.Where(l => l < 0) select q + 1; }" },
+                { "{     var e = from q in a.Where(condition) select q + 1; }", "{     var e = from q in a.Where(l => l < 0) select q + 1; }" },
+                { "var e = from q in a.Where(condition) select q + 1;", "var e = from q in a.Where(l => l < 0) select q + 1;" },
+                { "var e = from q in a.Where(condition) select q + 1", "var e = from q in a.Where(l => l < 0) select q + 1" },
+                { "e = from q in a.Where(condition) select q + 1", "e = from q in a.Where(l => l < 0) select q + 1" },
+                { "from q in a.Where(condition)", "from q in a.Where(l => l < 0)" },
+                { "select q + 1", "select q + 1" },  // select clause
+                { "select q + 1", "select q + 1" }   // query body
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions5()
+        {
+            var src1 = @"
+int a() 
+{
+    int b()
+    {
+        int c()
+        {
+            int d() { return 0; };
+        };
+
+        return c();
+    };
+
+    return b();
+}
+";
+            var src2 = @"
+int a() 
+{
+    int b()
+    {
+        int c()
+        {
+            int d() { return 0; };
+        };
+
+        return c();
+    };
+
+    return b();
+}
+";
+
+            var matches = GetMethodMatches(src1, src2);
+            var actual = ToMatchingPairs(matches);
+
+            var expected = new MatchingPairs
+            {
+                { "int a()  {     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }", "int a()  {     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }" },
```

Comment:
>, [](start = 167, length = 1)

I'd insert new lines after coma here and other long lines to make this more readable